### ISSUE TITLE
[Accessibility]: All text leaves are announced as clickable

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35267]
### Audit type

Screen reader navigation

### Problem description

See additional [slack context](https://sourcegraph.slack.com/archives/C038K3PR2M7/p1652192073605919).

Every text element in the app (it seems) announces itself as "clickable" in using VoiceOver. Every standard `text leaf` gets announced as `text element, clickable`.

Possible related issues/resources:

- https://github.com/facebook/react/issues/20895
- https://stackoverflow.com/questions/19702445/error-messages-reading-clickable-in-voice-over/24337647#24337647

### Expected behavior

Text isn't announced as clickable unless it actually does something. If this is something screen readers just _do_ then feel free to close this (it seems odd to me but maybe it's idiomatic usage patterns).

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
Assigning myself for further investigation

**Comment by sourcegraph-bot-2** [comment]
Heads up @taylorsperry @jasongornall - the "team/frontend-platform" label was applied to this issue.

**Comment by umpox** [comment]
So this doesn't happen in Chrome but I validated that this does happen in Firefox. It is _technically_ correct, but obviously isn't a good solution for screen reader users. It seems that this is best fixed on the browser level (like Chrome does), Firefox has an open bug for this here: https://bugzilla.mozilla.org/show_bug.cgi?id=1785268

I will close this issue as I don't believe there is much we can directly do here right now
